internal/controller/build/integrations/source/interfaces.go

```diff
@@ -0,0 +1,13 @@
+package source
+
+import "context"
+
+// SourceHandler is an interface that defines the operations that can be performed on the source provider
+// (GitHub/BitBucket/GitLab/etc.) by the build controller during the reconciliation process.
+type SourceHandler[T any] interface {
```

Comment:
```suggestion
type Handler[T any] interface {
```
The name `Handler` will be enough as we already in the source package so that users will access it as `source.Handler` instead of `source.SourceHandler`